bootspec/src/v1.rs

```diff
@@ -31,12 +34,16 @@ pub struct GenerationV1 {
     pub initrd: Option<PathBuf>,
     /// Path to "append-initrd-secrets" script -- $toplevel/append-initrd-secrets
     pub initrd_secrets: Option<PathBuf>,
-    /// Mapping of specialisation names to their boot.json
-    pub specialisation: HashMap<SpecialisationName, GenerationV1>,
     /// System double, e.g. x86_64-linux, for the system closure
     pub system: String,
+    /// Mapping of specialisation names to their boot.json
+    #[serde(default = "HashMap::new")]
+    pub specialisation: HashMap<SpecialisationName, GenerationV1>,
     /// config.system.build.toplevel path
     pub toplevel: SystemConfigurationRoot,
+    /// User extensions for this specification
+    #[serde(default = "HashMap::new", skip_serializing_if = "HashMap::is_empty")]
+    pub extensions: HashMap<String, Extension>,
```

Comment:
I do not think this is correct. This does not allow users to specify arbitrary extensions but forces them to use a nested set (or nested hash maps). Looking at bootspec.cue, however, arbitrary extensions are supposed to be supported. I guess you made this change to encourage namespacing, but afaik namespacing is not REQUIRED although encouraged. I think this would be correct: 

```suggestion
    pub extensions: Extension,
```

Reply:
We discussed this over Matrix here: https://matrix.to/#/!CDsPGxqljDUrcevkKh:grahamc.com/$vYmtI0SiSeLKIw2daz_gLq9jOZBVtygLkOCi6bp875c?via=grahamc.com&via=matrix.org&via=tchncs.de and concluded we want to amend the RFC to enforce namespacing.

`bootspec.cue` will support it too soon, I just did not have enough brain power to write the right spec.

Reply:
If you have a strong feeling against, we should postpone this PR until we sort out this in the RFC.

Reply:
Ah I missed that convo. I don't have a big issue with it, but I also don't see too much value in it. It doesn't really enforce namespacing (what we want) but forces people to use a nested set/object/struct/hashmap/dict/etc which can still be abused in weird ways. There also is some charm to a simple and plain `extensions` field (its what people know and expect). It's not a deal breaker but I have a weak opinion towards arbitrary extensions.

Reply:
IMHO, let's go with this as-is and let's bring this to the next RFC meeting as a point.

Reply:
Mentioned in RFC-0125.